### ISSUE TITLE
No declaration generated with components + decorators

### ISSUE THREAD

**Issue by dattebayorob** [wessberg/rollup-plugin-ts#38]
When a component wrapped by a decorator is exported, rollup-plugin-ts cant generate de .d.ts interface .
I noticed it with mobx-react observer, but with others decorator as well will reproduce same issue.

[Here](https://github.com/dattebayorob/lib-tests/commit/a9f1027a3e4c69bcafb80dcbd12e8e52dade055e#diff-e8acc63b1e238f3255c900eed37254b8) I created a component using withRouter from react-router-dom, and the output of index.d.ts is:
```
##dist/index.d.ts
import React from "react";
declare const Col: React.FC;
interface RowProps extends React.HTMLAttributes<HTMLDivElement> {
    children?: React.ReactNode;
}
declare const Row: React.FC<RowProps>;
export { Row, Col, Router };
```

When exporting component [without decorators](https://github.com/dattebayorob/lib-tests/commit/910919ee70689f3d80de82cfbe6cb92419ca1e1a), rollup-plugin-ts recognize types

```
##dist/index.d.ts
import React from "react";
declare const Col: React.FC;
interface RowProps extends React.HTMLAttributes<HTMLDivElement> {
    children?: React.ReactNode;
}
declare const Row: React.FC<RowProps>;
interface Props {
    test?: string;
}
declare const Router: React.FC<Props>;
export { Row, Col, Router };
```

rollup-plugin-ts version: 1.1.73
I'm using tsdx, let me now if this is the wrong place to report it.

**Comment by wessberg** [comment]
Hi there. Thanks for reporting this issue. I'm hoping you can provide me with a little more details, because I'm a little confused as to the _decorator_ part of the issue? I don't see any decorators in any of the linked commits.

**Comment by dattebayorob** [comment]
Ops, I'm sorry, here[here](https://github.com/dattebayorob/lib-tests/blob/a9f1027a3e4c69bcafb80dcbd12e8e52dade055e/lib/src/components/router/router.tsx#L16)

**Comment by dattebayorob** [comment]
In my case, the Router Props on .d.ts arent generated when a component decorated is exported.

The  rollup-plugin-typescript2 ignores decorators on type definitionss, and generate .d.ts with the props Interface only, but rollup

**Comment by wessberg** [comment]
> Ops, I'm sorry, here[here](https://github.com/dattebayorob/lib-tests/blob/a9f1027a3e4c69bcafb80dcbd12e8e52dade055e/lib/src/components/router/router.tsx#L16)

I still don't see any decorators 🙂 But I do see the export being assigned to the result of wrapping it in a function invocation. I'll look into it and set up a test case.

**Comment by dattebayorob** [comment]
Oh, thanks, in react ecossistem I know HOCs as decorator / annotations kkkkk sorry

**Comment by wessberg** [comment]
This issue has been fixed in [v1.1.74](https://github.com/wessberg/rollup-plugin-ts/releases/tag/v1.1.74)
